src/transformState.js

```diff
@@ -6,6 +6,26 @@
  */
 function transformState(state, actions) {
   // write code here
-}
+  const obj = state;
+
+  for (const key of actions) {
+    if (key.type === 'addProperties') {
+      Object.assign(obj, key.extraData);
+    }
+
+    if (key.type === 'removeProperties') {
+      for (const remove of key.keysToRemove) {
+        delete obj[remove];
+      }
+    }
 
+    if (key.type === 'clear') {
+      for (const properties in state) {
+        delete obj[properties];
+      }
+    }
```

Comment:
it'll be better to use `switch / case` instead of `if` here =)
![image](https://user-images.githubusercontent.com/105542865/189683476-72b022eb-91c3-4be8-ad7f-fbb14fa218a2.png)